app/grandchallenge/algorithms/forms.py

```diff
@@ -555,13 +556,34 @@ class AlgorithmImageUpdateForm(SaveFormInitMixin, ModelForm):
 
     class Meta:
         model = AlgorithmImage
-        fields = ("requires_gpu", "requires_memory_gb")
+        fields = ("requires_gpu", "requires_memory_gb", "comment")
         labels = {"requires_gpu": "GPU Supported"}
         help_texts = {
             "requires_gpu": "If true, inference jobs for this container will be assigned a GPU"
         }
 
 
+class ImageMarkDesiredForm(Form):
+    def __init__(self, *args, image, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.instance = image
+
+    def clean(self):
+        if not self.instance.is_manifest_valid:
+            raise ValidationError(
+                "Cannot mark an invalid image as active image."
+            )
+
+        if not self.instance.can_execute:
+            if self.instance.algorithm.image_upload_in_progress:
```

Comment:
`self.instance.algorithm.image_upload_in_progress` needs to be checked regardless of whether this particular image can be executed or not. Think of when a user uploads a new container, it's not validated yet, but they then try and activate an old one.